test/levels/Switch.test.js

```diff
@@ -0,0 +1,83 @@
+const Switch = artifacts.require('./levels/Switch.sol')
+const SwitchFactory = artifacts.require('./levels/SwitchFactory.sol')
+
+const Ethernaut = artifacts.require('./Ethernaut.sol')
+
+import * as utils from '../utils/TestUtils'
+import expectThrow from 'zeppelin-solidity/test/helpers/expectThrow'
+import toPromise from 'zeppelin-solidity/test/helpers/toPromise'
+
+contract('Switch', function(accounts) {
+
+  let ethernaut
+  let level
+  let instance
+  let player = accounts[0]
+
+  before(async function() {
+    ethernaut = await Ethernaut.new();
+    level = await SwitchFactory.new()
+    await ethernaut.registerLevel(level.address)
+    instance = await utils.createLevelInstance(
+      ethernaut, level.address, player, Switch,
+      {from: player}
+    )
+  });
+
+  describe('instance', function() {
+
+    it('should not be immediately solvable', async function() {
+
+      // make sure the factory fails
+      const ethCompleted = await utils.submitLevelInstance(
+        ethernaut,
+        level.address,
+        instance.address,
+        player
+      )
+      assert.equal(ethCompleted, false);
+    });
+
+    it('it should allow the user to solve the level', async function() {
+
+      // build the necessary function selectors
+      let flipSwitchSelector = web3.sha3("flipSwitch(bytes)").substring(2,10)
+      let onFunctionSelector = web3.sha3("turnSwitchOn()").substring(2,10)
+      let offFunctionSelector = web3.sha3("turnSwitchOff()").substring(2,10)
+
+      let dataPosition = '20' // hex for 32
+      let dataLength = 4 // _data only has a 4 byte length
+
+      // construct standard calldata
+      let calldata = flipSwitchSelector //flipSwitch function selector
+                     + "0".repeat(62) + dataPosition.toString() // hex position of _data
+                     + "0".repeat(63) + dataLength.toString() // length of _data
+                     + offFunctionSelector + "0".repeat(56) // The _data variable, containing the turnSwitchOff selector
+
+      // expect the turnSwitchOff function to not revert
+      await instance.sendTransaction({data: calldata});
+
```

Comment:
Could use another test: expect the `turnSwitchOn` function to revert on standard call

Reply:
Yep, agree.